### DISCUSSION TITLE
No longer working on latest obsidian. [Related to YAML properties editor in newer Obsidian releases]

### DISCUSSION THREAD

**Discussion by ambitendencies** [SebastianMC/obsidian-custom-sort#95] [Q&A]
My previous sortspec config doesn't work anymore on latest update.

**Comment by SebastianMC** [comment]
hi @ambitendencies !

to be able to help I need more details:
- platform (linux / mac / win / android / ios)
- exact version of Obsidian
- what exactly is not working? Is there any error message?
- if possible, please attach your sortspec, or relevant snippet of sortspec or an equivalent snippet/example if privacy matters

I'm using Obsidian on a daily basis, also the insider versions, custom sorting is working fine.

In the recent Obsidian public version (1.4.12) there is a feature (properties ui in editor) which can break the syntax of sortspec YAML (aka properties) in some cases (with nested sorting order specifications). This is because the Obsidian properties plugin automatically applies its own indentation, regardless of what you had there before (how many spaces). To see if this is your case I need more details, as described above

**Comment by TheOneLight** [comment]
On my system, too (Windows 11, Obsidian version 1.4.13, installer version 1.4.12), it effectively stopped working. 

At first, it still worked, but when I made changes, it stopped. However, making a change was already very difficult, as the entire row of folder titles which I had before in YAML now had been automatically transformed into a property called "sorting-spec". 

It seems that, particularly if there are more than a few folders named in the field for that, when one wants to change it later, it is not possible. I can only delete everything and fill it in anew, and I am not really sure how to do that. Could you perhaps give an example of a few folder names and how we actually have to handle this after the update?

I do still have the file "sortspec.md", at the same place as before. Then, there is the property "sorting-spec" now. What should I fill in in the field to the right? And how do I separate the names? 

Before, I would just write them in another row. But now, if there is a folder called e.g. "this is a foldername", and then there is another folder called "this is folder 2", it now would all end up in one row "this is a foldername this is folder 2". And, if filling in enough of these names, I suddenly cannot edit that field any longer.

I could imagine that you might find more users with the same problem, but I do not know. I would, of course, appreciate your advice.

**Comment by SebastianMC** [comment]
Hi @TheOneLight,

thanks for your detailed input!

To bring more clarity to the new (and confusing) property editor of Obsidian I've created a dedicated documentation page here: [How to create or edit a multi-valued property in Obsidian](https://github.com/SebastianMC/obsidian-custom-sort/wiki/How-to-create-or-edit-a-multi%E2%80%90line-property-in-Obsidian%3F)

I tried to keep the page simple, hoping the materials are sufficient to help.

Could you please look at that documentation page?
You are using Windows - could you confirm, if `shift-enter` inserts the line break also on Windows?
Out of curiosity, if you examine the three sections of the page:
- creating multi-line property in YAML properties editor
- editing multi-line property in YAML properties editor
- editing multi-line property in _source view mode_

would you prefer the _source view mode_ or the new YAML properties editor?

Thanks in advance for your feedback!

**Comment by TheOneLight** [comment]
Hi *[SebastianMC](https://github.com/SebastianMC), thank you very much. I had not known about shift+enter, with that I could edit it, and sorting works again. 

(Perhaps you could link that information more prominently from the homepage of the plugin? I would have found integrating some sample folder names or file names in the instruction very helpful for the non-technically versed user. Even as the plugin homepage on Github is at the moment, it would have helped me, when I started out, to see instead of only

> ---
> sorting-spec: |
>   order-desc: a-z
> ---   
>

additionally something like

> ---
> sorting-spec: |
>My first folder name
>My second folder name
>My first file name   
> ---   
>

etc.

[My impression is that sometimes programmers think some things are just obvious, whereas those users who are not familiar with programming and more or less "wading through" things only half knowing what they do at best, might, then, give up at some stage, where it would have only been a short sentence or explanation to add to make it clear to them.

I am not specifically speaking about you or your plugin, but more generally about how I have felt sometimes when trying to change something on one of my websites and having to go into CSS, which I never learned, sometimes just wishing one sentence more here and there, one more extensive example with explanations... (But maybe others find it quicker to go through when it is explained with less words, I do not know.)

Coming from that perspective:  You could now bring what I suggested above together with the helpful link you provided in your previous post. I believe users without a technical background might find it easier, then.])

A for your question: After the update, I had actually looked for something like the source view mode, assuming something like that must be there, but somehow I had overlooked it. Now I have found it, thank you for mentioning it. When I now compare editing the properties or the source view mode, I can work with both, but I guess using the source view mode for that in the case of more extensive property content like with larger lists of folder or file names in the case of your plugin might appear to be a bit more convenient to me. With only one line to edit, I might have preferred the non-source view mode, but the more there is to edit, the more convenient the source view mode might be for me.

Thank you, by the way, for creating the plugin. It is very helpful.

**Comment by SebastianMC** [comment]
Hi @TheOneLight 

I fully get your perspective! Thanks for presenting it, I appreciate!

Creating a good documentation for a non-trivial product is a challenge and a work on its own.
I started #66 to create a quality documentation. Yet, it turns out I don't have enough time to focus both on the plugins itself and on documentation. This is a private project, done after hours in spare time. A trade-off, so far, resulted on focus on functionality first. And a major update - drag & drop reordering support via Bookmarks core functionality integration is awaiting its time slot.

Following your suggestion, I will add the link to wiki/videos on the main documentation page, this is a quick task, can be helpful.

BTW @ambitendencies (the original reporter of this thread) I will turn this issue into a discussion page in a dedicated section (a link will follow here). It looks it can be helpful for others

**Comment by SebastianMC** [comment]
Hi @TheOneLight 

FYI your two suggestions materialised on the main documentation page [README.md](https://github.com/SebastianMC/obsidian-custom-sort/blob/master/README.md)

**Reply by TheOneLight** [reply]
Thank you, @SebastianMC!
